core/src/main/java/org/axonframework/queryhandling/SimpleQueryBus.java

```diff
@@ -188,6 +217,120 @@ public SimpleQueryBus(MessageMonitor<? super QueryMessage<?, ?>> messageMonitor,
+                                                                                  .getSubscriptionQueryHandler(),
+                                                                          updateHandler);
+                            invocationSuccess = true;
+                        }
+                    } else {
+                        invokeRegularQueryHandler(uow, subscription.getQueryHandler(), updateHandler);
+                        invocationSuccess = true;
+                    }
+                } catch (NoHandlerForQueryException e) {
+                    // Ignore this Query Handler, as we may have another one which is suitable
+                }
+            }
+            if (!invocationSuccess) {
+                throw new NoHandlerForQueryException(
+                        format("No suitable handler was found for %s with response type %s and update type %s",
+                               interceptedQuery.getQueryName(),
+                               interceptedQuery.getResponseType(),
+                               interceptedQuery.getUpdateResponseType()));
+            }
+        } catch (Exception e) {
+            monitorCallback.reportFailure(e);
+            updateHandler.onError(e);
+        }
+
+        return registration;
+    }
+
+    /**
+     * Invokes regular query handler and completes the update handler right afterwards.
+     *
+     * @param uow           the Unit of Work in which the query handler will be invoked
+     * @param queryHandler  the query handler to be invoked
+     * @param updateHandler the update handler to be invoked with result of query handler
+     * @param <Q>           the query type
+     * @param <I>           the initial result type
+     * @throws Exception propagated from query handler
+     */
+    @SuppressWarnings("unchecked")
+    private <Q, I> void invokeRegularQueryHandler(UnitOfWork<QueryMessage<Q, I>> uow,
+                                                     MessageHandler<? super QueryMessage<?, I>> queryHandler,
+                                                     UpdateHandler<I, ?> updateHandler) throws Exception {
+        I initialResult = interceptAndInvoke(uow, queryHandler).getPayload();
+        updateHandler.onInitialResult(initialResult);
+        updateHandler.onCompleted();
+    }
+
+    /**
+     * Invokes subscription query handler with freshly initialized emitter.
+     *
+     * @param uow           the Unit of Work in which the query handler will be invoked
+     * @param queryHandler  the query handler to be invoked
+     * @param updateHandler the update handler to be invoked with result of query handler
+     * @param <Q>           the query type
+     * @param <I>           the initial result type
+     * @param <U>           the incremental update type
+     * @return handle to cancel updates on this query
+     *
+     * @throws Exception propagated from query handler
+     */
+    @SuppressWarnings("unchecked")
```

Comment:
What line it's warning is being suppressed?

Reply:
`SubscriptionQueryMessageHandler` is of a raw type, will fix it.